### ISSUE TITLE
Site Migration: handle email unverified when selecting free trial

### ISSUE THREAD

**Issue by andres-blanco** [Automattic/wp-calypso#87735]
When a Free trial is selected a check is being made to see if the user has a verified email. If not we should handle that, probably by reusing the existing VerifyEmail component from the current import flow.

**Comment by gabrielcaires** [comment]
Do we have to have it before the call for testing, since we will not be available for all users on our first release?

**Comment by andres-blanco** [comment]
I think new users are a good portion of this flow, so we should probably have it in place, otherwise, once we deploy this, we would need to add a check and remove the free trial option. I will tackle this.

**Comment by andres-blanco** [comment]
This is something that should be done, but probably can wait, as Gabriel pointed out.

**Comment by markbiek** [comment]
Just noting that the `/wpcom/v2/sites/:siteID/hosting/trial/check-eligibility/wp_bundle_migration_trial_monthly` endpoint is called when the "Plan Upgrade" step loads.

And the payload looks like this:

```
{
    "body": {
        "blog_id": 231564240,
        "eligible": false,
        "error_code": "email-unverified",
        "message": "You must verify your email before starting the trial."
    },
    "status": 200,
    "headers": {
        "Allow": "GET"
    }
}
```

**Comment by markbiek** [comment]
Noting that we've shipped an initial version where we send the user to the "Verify Email" step, then back to "Plan Upgrade".

A future enhancement could be automatically adding the free trial once "Verify Email" is complete and redirecting to "Bundle Upgrade" so the user doesn't have to see the "Plan Upgrade" step twice.

**Comment by andres-blanco** [comment]
I am closing this in favor of https://github.com/Automattic/wp-calypso/issues/89254 where we track specifically the automatic addition of the trial.
